index.html

```diff
@@ -2520,6 +2519,21 @@ <h3><dfn>Writing content</dfn></h3>
                 If |source|'s records [= list/is empty =], [= exception/throw =]
                 a {{TypeError}} and abort these steps.
               </li>
+              <li>
+                Increment |recordsDepth| by one.
+              </li>
+              <li>
+                If |recordsDepth| > `2`, [= exception/throw =] a {{TypeError}}
+                and abort these steps.
+                <p class="issue" data-number="620">
+                  To prevent infinite loops, Web IDL bans dictionaries from
+                  including themselves as members. As the `data` dictionaries
+                  will have to have been created by a developer before
+                  initiating the function call, these will in practice be
+                  finite, so we work around this by using `any` as the
```

Comment:
There are tricky ways that a developer can craft an object that changes as the algorithm processes it. What I would say here instead is that for practical purposes there is only so much room for storing an NDEF message and so we can apply a limit on the depth of nested messages which developers are unlikely to ever hit.

Reply:
Applied